app/ante/evm/08_vesting_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+package evm_test
+
+import (
+	"math/big"
+	"time"
+
+	"cosmossdk.io/math"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	sdkvesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	"github.com/evmos/evmos/v16/app/ante/evm"
+	"github.com/evmos/evmos/v16/testutil/integration/evmos/grpc"
+	testkeyring "github.com/evmos/evmos/v16/testutil/integration/evmos/keyring"
+	"github.com/evmos/evmos/v16/testutil/integration/evmos/network"
+	vestingtypes "github.com/evmos/evmos/v16/x/vesting/types"
+)
+
+type AccountExpenses = map[string]*evm.EthVestingExpenseTracker
+
+func (suite *EvmAnteTestSuite) TestCheckVesting() {
+	keyring := testkeyring.New(1)
+	unitNetwork := network.NewUnitTestNetwork(
+		network.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+	)
+	grpcHandler := grpc.NewIntegrationHandler(unitNetwork)
+	sender := keyring.GetAccAddr(0)
+	addedExpense := math.NewInt(100)
+
+	testCases := []struct {
+		name                  string
+		expectedError         error
+		getAccountAndExpenses func() (authtypes.AccountI, AccountExpenses)
+	}{
+		{
+			name:          "success: non clawback account should be successful",
```

Comment:
there are some more generic test cases like "invalid from address" on the original tests, are they covered elsewhere already?

Reply:
This cases were covered by the decorator not the check function itself. That will be covered in the integration tests instead